controller/telemetry/server.go

```diff
@@ -184,29 +184,52 @@ func NewServer(addr, prometheusUrl string, ignoredNamespaces []string, kubeconfi
 func (s *server) Query(ctx context.Context, req *read.QueryRequest) (*read.QueryResponse, error) {
```

Comment:
Much of the behavior of this method switches depending on if start/stop/step is set or not.  Would it make sense to split this into two methods, `QueryTs` and `QuerySummary` or something like that?

Reply:
I agree, now that the telemetry service's `Query()` endpoint is supporting two distinct use cases, this warrants modifying the API to be more explicit about it. This will require a protobuf change in the telemetry service, which is better done in a separate PR, I've filed #312 to track.